waspc/src/Analyzer/Parser/Lexer.x

```diff
@@ -0,0 +1,139 @@
+{
+-- This file is processed by Alex (https://www.haskell.org/alex/) and generates
+-- the module `Analyzer.Parser.Lexer`
+
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Analyzer.Parser.Lexer
+  ( lexer
+  ) where
+
+import Analyzer.Parser.Util (ParserInput, Parser, ParserState (..), updatePosition, putInput, setStartCode)
+import Analyzer.Parser.Token (Token (..), TokenClass (..))
+import Analyzer.Parser.ParseError (ParseError (..))
+import Control.Monad.Trans.State.Lazy (get)
+import Control.Monad.Trans.Except (throwE)
+import Control.Monad.Trans.Class (lift)
+import Data.Word (Word8)
+import Codec.Binary.UTF8.String (encodeChar)
+
+import Debug.Trace
+}
+
+-- Character set aliases
+$digit = 0-9
+$alpha = [a-zA-Z]
+$identstart = [_$alpha]
+$ident = [_$alpha$digit]
+$any = [.$white]
+
+-- Regular expression aliases
+@string = \"([^\\\"]|\\.)*\" -- matches string-literal on a single line, from https://stackoverflow.com/a/9260547/3902376
+@double = "-"? $digit+ "." $digit+
+@integer = "-"? $digit+
+@ident = $identstart $ident* "'"*
+
+tokens :-
+
+-- Skips whitespace
+<0>       $white+ ;
+
+-- Quoter rules
+<0>       "{=" @ident { beginQuoter }
+<quoter>  @ident "=}" { endQuoter }
```

Comment:
Maybe you cold provide a comment here, explaining how these start codes help, how is quoter parsing working thanks to that? I actually don't understand how that works right now so it would be helpful! If not here then below where beingQuoter and endQuoter are defined.